plugins/formal-verification/testData/diagnostics/no_contracts/predicates_access.fir.diag.txt

```diff
@@ -0,0 +1,218 @@
+/predicates_access.kt:(223,246): info: Generated Viper text for accessSuperTypeProperty:
+field class_A$member_a: Int
+
+field class_B$member_b: Int
+
+field class_C$member_x: Ref
+
+field class_C$member_y: Ref
+
+function T_class_global$class_A$get$class_A$member_a(special$get$function$subject: Ref): Int
+  requires acc(T_class_global$class_A(special$get$function$subject), wildcard)
+{
+  (unfolding acc(T_class_global$class_A(special$get$function$subject), wildcard) in
+    special$get$function$subject.class_A$member_a)
+}
```

Comment:
Do we still want to generate these functions?  You don't seem to be using them.

Reply:
This is one of the things I want to discuss on Monday. I think these functions will never be good enough because of recursive types. Probably they are going to be replaced by mapping kotlin nested accesses to nested unfolding expressions when we need a single expression for a nested access (e.g. in pre/postconditions)